File: Python/stage1/S1_SW_03.py
# S1_SW_03 — Simple calculator
def calc(a, op, b):
   match op:
       case "+":
           return a + b
       case "-":
           return a - b
       case "*":
           return a * b
       case "/":
           if b == 0:
               return None
           return a / b
       case _:
           return None
# Tests
tests = [
   (4, "+", 2),
   (4, "-", 2),
   (4, "*", 2),
   (4, "/", 2),
   (4, "/", 0),
]
for t in tests:
   print(t, "=>", calc(*t))

File: js/stage1/S1_SW_03.js
// S1_SW_03 — Simple calculator
function calc(a, op, b) {
  switch (op) {
    case "+":
      return a + b;
    case "-":
      return a - b;
    case "*":
      return a * b;
    case "/":
      return b === 0 ? null : a / b;
    default:
      return null;
  }
}
// Tests
console.log(calc(5, "+", 3)); // 8
console.log(calc(5, "-", 3)); // 2
console.log(calc(5, "*", 3)); // 15
console.log(calc(5, "/", 0)); // null
console.log(calc(5, "x", 3)); // null
